recirq/time_crystals/dtcsimulation.py

```diff
@@ -0,0 +1,353 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Tuple, List, Iterator
+
+import cirq
+import functools
+from recirq.time_crystals.dtcexperiment import DTCExperiment, comparison_experiments
+import numpy as np
+import sympy as sp
+
+
+def symbolic_dtc_circuit_list(
+    qubits: Sequence[cirq.Qid], cycles: int
+) -> List[cirq.Circuit]:
+
+    """Create a list of symbolically parameterized dtc circuits, with increasing cycles
+
+    Args:
+        qubits: ordered sequence of available qubits, which are connected in a chain
+        cycles: maximum number of cycles to generate up to
+
+    Returns:
+        list of circuits with `0, 1, 2, ... cycles` many cycles
+
+    """
+
+    num_qubits = len(qubits)
+
+    # Symbol for g
+    g_value = sp.Symbol("g")
+
+    # Symbols for random variance (h) and initial state, one per qubit
+    local_fields = sp.symbols(f"local_field_:{num_qubits}")
+    initial_state = sp.symbols(f"initial_state_:{num_qubits}")
+
+    # Symbols used for PhasedFsimGate, one for every qubit pair in the chain
+    thetas = sp.symbols(f"theta_:{num_qubits - 1}")
+    zetas = sp.symbols(f"zeta_:{num_qubits - 1}")
+    chis = sp.symbols(f"chi_:{num_qubits - 1}")
+    gammas = sp.symbols(f"gamma_:{num_qubits - 1}")
+    phis = sp.symbols(f"phi_:{num_qubits - 1}")
+
+    # Initial moment of Y gates, conditioned on initial state
+    initial_operations = cirq.Moment(
+        [cirq.Y(qubit) ** initial_state[index] for index, qubit in enumerate(qubits)]
+    )
+
+    # First component of U cycle, a moment of XZ gates.
+    sequence_operations = []
+    for index, qubit in enumerate(qubits):
+        sequence_operations.append(
+            cirq.PhasedXZGate(
+                x_exponent=g_value,
+                axis_phase_exponent=0.0,
+                z_exponent=local_fields[index],
+            )(qubit)
+        )
+
+    # Initialize U cycle
+    u_cycle = [cirq.Moment(sequence_operations)]
+
+    # Second and third components of U cycle, a chain of 2-qubit PhasedFSim gates
+    #   The first component is all the 2-qubit PhasedFSim gates starting on even qubits
+    #   The second component is the 2-qubit gates starting on odd qubits
+    operation_list = []
+    other_operation_list = []
+    previous_qubit = None
+    previous_index = None
+    for index, qubit in enumerate(qubits):
+        if previous_qubit is None:
+            previous_qubit = qubit
+            previous_index = index
+            continue
+
+        # Add an fsim gate
+        coupling_gate = cirq.ops.PhasedFSimGate(
+            theta=thetas[previous_index],
+            zeta=zetas[previous_index],
+            chi=chis[previous_index],
+            gamma=gammas[previous_index],
+            phi=phis[previous_index],
+        )
+        operation_list.append(coupling_gate.on(previous_qubit, qubit))
+
+        # Swap the operation lists, to avoid two-qubit gate overlap
```

Comment:
I was trying to avoid index math for more pythonic style, and I think I found a way to do that and remove the "previous qubit" update lines (by packaging them into the `for` statement). 

However, if the number of qubits is odd, then we need an odd number of coupling gates, so it doesn't make sense (I think) to iterate by two instead of one because it would add a clunky check to prevent an out of bounds error at the end of the list. 

This approach still needs to swap the moments, in order to alternate which one it adds the gate to. Originally this was done with pair assignment (`current_moment, other_moment = other_moment, current_moment`). With the "only one variable assignment per line" restriction, three lines are necessary.